backend/roles/chat-roles-manager.js

```diff
@@ -0,0 +1,133 @@
+"use strict";
+
+const accountAccess = require("../common/account-access");
+const twitchApi = require("../twitch-api/client");
+const logger = require("../logwrapper");
+const NodeCache = require("node-cache");
+
+const userRoleCache = new NodeCache({ stdTTL: 30, checkperiod: 5 });
+
+const isBroadcaster = (username) => {
+    return username === accountAccess.getAccounts().streamer.username;
+};
+
+const isModerator = async (userId) => {
+    try {
+        const client = twitchApi.getClient();
+        const moderators = await client.helix.moderation.getModeratorsPaginated(accountAccess.getAccounts().streamer.userId).getAll();
+
+        if (moderators) {
+            for (const moderator of moderators) {
+                if (moderator.userId === userId) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    } catch (error) {
+        logger.debug("Failed to get moderators", error);
+        return false;
+    }
+};
+
+const isVip = async (username) => {
+    try {
+        const twitchChat = require("../chat/twitch-chat");
+        const vips = await twitchChat.getVips();
+
+        if (vips) {
+            for (const vip of vips) {
+                if (vip === username) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
```

Comment:
We can shorten this similarly to the mods suggestion:
```js
return vips ? vips.includes(username) : false;
```

Reply:
Fixed.